firebase_admin/db.py

```diff
@@ -125,40 +125,75 @@ def child(self, path):
 
-    def get(self):
-        """Returns the value at the current location of the database.
+    def etag(self):
+        """Returns the ETag at the current location of the database.
+        Returns:
+            str: ETag of the current database Reference.
+        """
+        headers = self._client.request('get', self._add_suffix(),
+                                       headers={'X-Firebase-ETag' : 'true'},
+                                       resp_headers=True,
+                                       params='print=silent')
+        return headers.get('ETag')
+
+    def get(self, etag=False):
+        """Returns the value, and possibly the ETag, at the current location of the database.
 
         Returns:
-          object: Decoded JSON value of the current database Reference.
+          object: Decoded JSON value of the current database Reference if etag=False, otherwise
+                  the decoded JSON value and the corresponding ETag.
 
         Raises:
           ApiCallError: If an error occurs while communicating with the remote database server.
         """
-        return self._client.request('get', self._add_suffix())
-
-    def _get_with_etag(self):
-        """Returns the value at the current location of the database, along with its ETag."""
-        data, headers = self._client.request(
-            'get', self._add_suffix(), headers={'X-Firebase-ETag' : 'true'}, resp_headers=True)
-        etag = headers.get('ETag')
-        return etag, data
+        if etag:
+            data, headers = self._client.request('get', self._add_suffix(),
+                                                 headers={'X-Firebase-ETag' : 'true'},
+                                                 resp_headers=True)
+            etag = headers.get('ETag')
+            return data, etag
+        else:
+            return self._client.request('get', self._add_suffix())
 
-    def set(self, value):
+    def set(self, value, etag=None):
         """Sets the data at this location to the given value.
-
         The value must be JSON-serializable and not None.
+        If etag != None, checks to make sure that the ETag value matches that of the location.
 
         Args:
           value: JSON-serialable value to be set at this location.
+          etag: Value of ETag that we want to check.
+
+        Returns:
+          object: Tuple of False, current location's etag, and snapshot of location's data
+                  if passed in etag does not match. 
 
         Raises:
-          ValueError: If the value is None.
+          ValueError: If the value is None, or if etag is not a string.
           TypeError: If the value is not JSON-serializable.
-          ApiCallError: If an error occurs while communicating with the remote database server.
+          ApiCallError: If an error occurs while communicating with the remote database server,
+                        or if the ETag does not match.
```

Comment:
Indent by 4 spaces. Might want to add a lint ignore directive here to fix the 2nd lint error.

`# pylint: disable=missing-raises-doc`

Might have to experiment a bit to get the placement of the directive right.